tools/cmake/FindBpfProg.cmake

```diff
@@ -0,0 +1,182 @@
+# SPDX-License-Identifier: GPL-2.0 OR BSD-3-Clause
+
+#[=======================================================================[.rst:
+FindBpfProg
+--------
+
+Find BpfProg
+
+This module finds if all the dependencies for eBPF Compile-Once-Run-Everywhere
+programs are available and where all the components are located.
+
+The caller may set the following variables to disable automatic
+search/processing for the associated component:
+
+  ``BPFTOOL_EXE``
+    Path to ``bpftool`` binary
+
+  ``CLANG_EXE``
+    Path to ``clang`` binary
+
+  ``LIBBPF_INCLUDE_DIRS``
+    Path to ``libbpf`` development headers
+
+  ``LIBBPF_LIBRARIES``
+    Path to `libbpf` library
+
+  ``GENERATED_VMLINUX``
+    Path to ``bpftool`` generated ``vmlinux.h`` file. If unset, this module will
+    attempt to automatically generate a copy.
+
+This module sets the following result variables:
+
+::
+
+  BpfProg_FOUND             = TRUE if all components are found
+
+
+This module also provides the ``bpf_prog()`` macro. This macro generates a
+cmake interface library for the BPF program's generated skeleton as well
+as the associated dependencies.
+
+.. code-block:: cmake
+
+  bpf_prog(<name> <source>)
```

Comment:
can you make it it ready for BPF static linking and allow specifying multiple source files, which will then be linked with `bpftool gen object` command?

Reply:
Yeah, we can extend using optional macro arguments `${ARGN}` ( https://cmake.org/cmake/help/latest/command/macro.html#arguments ).

Then in bpf_prog() macro, we can do something like (in pseudocode):
```
for optsrc in ${ARGN} + input:
  # build the .bpf.o

# run bpftool gen object <objs>

# build skeleton
```

Is the static linking stuff landed upstream yet?

Reply:
> Is the static linking stuff landed upstream yet?

Not completely. Two more patch sets are in review right now.

Reply:
Ok, we can extend it later then.